upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -197,3 +199,30 @@ func (tf *TemplateFunctions) ExternalDnsArgv() ([]string, error) {
 
 	return argv, nil
 }
+
+func (tf *TemplateFunctions) ProxyEnv() (map[string]string, error) {
+	envs := map[string]string{}
+	proxies := tf.cluster.Spec.EgressProxy
+	if proxies == nil {
+		return envs, nil
+	}
+	httpProxy := proxies.HTTPProxy
+	if httpProxy.Host != "" {
+		url := "http://"
+		if httpProxy.User != "" {
```

Comment:
Can we pull the user stuff?  Can we check if the hostname starts with http don't add it?  Will help to not have errors.